### PR TITLE
add integer divison for symints

### PR DESCRIPTION
**Pull request by Krovatkin**
### Description
This PR brings integer division (floor) to symints + tests.

### Issue

https://github.com/orgs/pytorch/projects/17/views/2

### Testing
added two tests to TestPySymInts

### CODE REVIEW

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -1223,10 +1227,28 @@ void initJITBindings(PyObject* module) {
...
+          [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
+            auto snb = toSymIntNode(a, b);
+            return a->floordiv(snb);
+          })
+      .def(
+          "__rfloordiv__",
           [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
             auto snb = toSymIntNode(a, b);
-            return a->div(snb);
+            return a->floordiv(snb);
```
Comment (ezyang) [inline torch/csrc/jit/python/init.cpp:1251]: The r rules are wrong; the operands are swapped.

**Comment by ezyang** [comment]
I'd advise adding a test for the r behavior, at least, as it's buggy here.

**Comment by Krovatkin** [comment]
> I'd advise adding a test for the r behavior, at least, as it's buggy here.

@ezyang uh oh ... lemme check...

at least this simple test always puts a SymInt argument first.

```
class Int:
    def __floordiv__(self, obj):
        print("__floordiv__")
        return Int()

    def __rfloordiv__(self, obj):
        print(f"{type(self).__name__} {type(obj).__name__}")
        return None

2 // Int()
```

```
Int int
```

**Comment by ezyang** [comment]
that's my point, you get called with (right, left) and if you're going to implement it with the non-flipped method you have to swap 'em.

**Comment by Krovatkin** [comment]
@ezyang 💯

File: test/test_dynamic_shapes.py [hunk 2]
```diff
@@ -149,6 +149,17 @@ def test_arith_ops(self):
...
 
+    @skipIfNoSympy
+    def test_reverse_arith_ops(self):
+        shape_env = ShapeEnv()
+
+        a = shape_env.create_symint("s1", 2)
+        self.assertTrue(5 // a == 5 // 2)
+
+        a = shape_env.create_symint("s1", 2)
+        self.assertTrue(5 * a == 5 * 2)
```
Comment (ezyang) [inline test/test_dynamic_shapes.py:160]: this test doesn't work cuz multiplication is commutative lol

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**